internal/pkg/runtime/engines/singularity/container_linux.go

```diff
@@ -531,6 +531,15 @@ func (c *container) mountGeneric(mnt *mount.Point) (err error) {
 		}
 	}
 	_, err = c.rpcOps.Mount(source, dest, mnt.Type, flags, optsString)
+	// when using user namespace we always try to apply mount flags,
+	// then if we get a permission denied error, we contine execution
+	// by ignoring the error and warn user
+	if remount && err != nil && c.userNS {
+		if err.Error() == syscall.Errno(syscall.EPERM).Error() {
+			sylog.Warningf("Could not remount %s: %s", dest, err)
```

Comment:
I don't think this should be a warning; verbose should be enough (and matching the behavior of 2.6), unless perhaps the remount is because the user explicitly asked for it with a :ro in a --bind option.

Are there any reasons that err might be EPERM other than a remount failure?

Reply:
For `EPERM` condition, this is a pretty defensive approach instead of ignoring all error, other errors could be caused by a dev mistake or something irregular